Implement bruteforce solution to graph 3-coloring

smanilov opened the pull request:
Have a look at the README.txt.

Compile with

```
g++ -O3 color.cpp
```

Run with

```
./a.out < 22-44-graph.txt
```

smanilov committed:
feat:Add bruteforce graph coloring solver

smanilov committed:
feat: Parallelize bruteforce GCS

Syndamia commented:
I've ran some tests and the C++ algorithm seems to work. However, you still need to clear the non-bruteforce files as well as improve documentation. I can do those for you, if you want.

Syndamia committed:
feat(graph-coloring/bruteforce): Proper TOML output and flake dependency

Syndamia committed:
feat(graph-coloring): Add bruteforce Makefile targets and information in README

smanilov committed:
feat: Add hint to README.md

The `greedy-gen` solution can be used to generate a random graph which
can then be used as an input to `bruteforce-gen`.

smanilov commented:
> I've ran some tests and the C++ algorithm seems to work. However, you still need to clear the non-bruteforce files as well as improve documentation. I can do those for you, if you want.

Not sure what you mean here by the "non-bruteforce" files. If you mean "greedy-*.txt", these are added intentionally, as they are referenced by the `README.md` in the `bruteforce-gen` dir.

Update the main `README.md` some more, so I think this is now ready to be merged.

PTAL.
